submissions/IhorOzerov/frogger/app.js

```diff
@@ -0,0 +1,113 @@
+const config = {
+    CANVAS_WIDTH : 505,
+    REFRESH_TIME : 500,
+    Y_START : 0,
+    X_START : 0,
+
+    ENEMY_CONST: {
+        refresh: -101,
+        locations: [62, 144, 228],
+        speed: (Math.random() * 200) + 200,
+        sprite: "images/enemy-bug.png"
+    },
+
+    PLAYER_CONST: {
+        x: 203,
+        y: 390,
+        stepX: 102,
+        stepY: 83,
+        sprite: "images/char-boy.png",
+        height: 60,
+        width: 70,
+        rightBorder: 400
+    }
+};
+
+const Enemy = function (x, y) {
+    this.x = x;
+    this.y = y;
+    this.speed = config.ENEMY_CONST.speed;
+    this.sprite = config.ENEMY_CONST.sprite;
+}; 
+
+Enemy.prototype.update = function (dt) {
+    this.x += config.ENEMY_CONST.speed * dt;
+    if (this.x > config.CANVAS_WIDTH) {
+        this.x = config.ENEMY_CONST.refresh;
+    };
+    this.checkCollision();
+};
+
+Enemy.prototype.checkCollision = function () {
+    
+    const enemyWidth = this.x + config.PLAYER_CONST.width;
+    const enemyHeight = this.y + config.PLAYER_CONST.height;
+    const playerWidth = player.x + config.PLAYER_CONST.width;
+    const playerHeight = player.y + config.PLAYER_CONST.height;
+    
+    if (player.x < enemyWidth &&
+        playerWidth > this.x &&
+        player.y < enemyHeight &&
+        playerHeight > this.y) {
+            player.x = config.PLAYER_CONST.x;
+            player.y = config.PLAYER_CONST.y;
```

Comment:
Requirements posted/linked by bot addreess specically issue we have in this fragment.
Please identify the requirement, post it here as a comment and fix code accordingly.

Reply:
`Classes do not refer to any global variables, like global variable player, which is an instance of Player class (referring to global constants and globals provided by the gaming platform like Resources is OK); Hint: pass instance of a game object (or objects) as an argument to other game objects they need to interact with.`

Reply:
already fixed it, check please